Compilation error

Hello there,

Im trying to compile this in Visual Studio 2017, Im installer msgpack and all dependencies with vcpkg but is still saying that the msgpack is not found.

 Add the installation prefix of "msgpack" to CMAKE_PREFIX_PATH or set
  "msgpack_DIR" to a directory containing one of the above files.  If
  "msgpack" provides a separate development package or SDK, be sure it has
  been installed.

As you said, the answer is no, here is the log

C:\PROGRAM FILES (X86)\MICROSOFT VISUAL STUDIO\2017\COMMUNITY\COMMON7\IDE\COMMONEXTENSIONS\MICROSOFT\CMAKE\CMake\bin\cmake.exe  -G "Ninja" -DCMAKE_INSTALL_PREFIX:PATH="C:\Users\Donato\CMakeBuilds\f242219b-f993-b533-bb49-e9263402c529\install\x64-Debug (default)"  -DCMAKE_CXX_COMPILER="C:/Program Files (x86)/Microsoft Visual Studio/2017/Community/VC/Tools/MSVC/14.16.27023/bin/HostX64/x64/cl.exe"  -DCMAKE_C_COMPILER="C:/Program Files (x86)/Microsoft Visual Studio/2017/Community/VC/Tools/MSVC/14.16.27023/bin/HostX64/x64/cl.exe"  -DCMAKE_BUILD_TYPE="Debug" -DCMAKE_MAKE_PROGRAM="C:\PROGRAM FILES (X86)\MICROSOFT VISUAL STUDIO\2017\COMMUNITY\COMMON7\IDE\COMMONEXTENSIONS\MICROSOFT\CMAKE\Ninja\ninja.exe" "C:\Users\Donato\bylihis_joblogger" returned with exit code: 1
CMake Error at plugin/CMakeLists.txt:17 (find_package):
  By not providing "Findmsgpack.cmake" in CMAKE_MODULE_PATH this project has
  asked CMake to find a package configuration file provided by "msgpack", but
  CMake did not find one.
  Could not find a package configuration file provided by "msgpack" with any
  of the following names:
    msgpackConfig.cmake
    msgpack-config.cmake
  Add the installation prefix of "msgpack" to CMAKE_PREFIX_PATH or set
  "msgpack_DIR" to a directory containing one of the above files.  If
  "msgpack" provides a separate development package or SDK, be sure it has
  been installed.	plugin	C:\Users\Donato\bylihis_joblogger\plugin/CMakeLists.txt	17	

But I don't know where to put that variable "CMAKE_TOOLCHAIN_FILE"

I did all the things but is still not working with visual studio on windows, same error about msgpack.

Other thing, have more luck doing this on debian, I put the "cmake ." and then the "make" but an error appears:
/root/dev/bylihis_joblogger/ets2-job-logger/plugin/include/jobplugin/PluginDefs.h:197:26: error: conversion from ‘const uint64_t’ {aka ‘const long unsigned int’} to ‘Json::Value’ is ambiguous
         root["income"] = income;

Thank you, I tried with VS and now the things are going better, but at the end one error and two warnings appears:

ERROR1:
C:\Program Files (x86)\Microsoft Visual Studio\2017\Community\Common7\IDE\MSBUILD : error MSB1016: Specify the verbosity level.
  Switch: -v

The CMakeSettings.json contain:
{
  "configurations": [
    {
      "name": "x64-Release",
      "generator": "Visual Studio 15 2017 Win64",
      "configurationType": "Release",
      "inheritEnvironments": [
        "msvc_x64_x64"
      ],
      "buildRoot": "${env.USERPROFILE}\\CMakeBuilds\\${workspaceHash}\\build\\${name}",
      "installRoot": "${env.USERPROFILE}\\CMakeBuilds\\${workspaceHash}\\install\\${name}",
      "cmakeCommandArgs": "",
      "buildCommandArgs": "-v",
      "ctestCommandArgs": ""
    }
  ]
}



----------------------------------------------

 CMake Warning at C:/Program Files (x86)/Microsoft Visual Studio/2017/Community/Common7/IDE/CommonExtensions/Microsoft/CMake/CMake/share/cmake-3.12/Modules/FindBoost.cmake:847 (message):
1>   New Boost version may have incorrect or missing dependencies and imported
1>   targets
1> Call Stack (most recent call first):
1>   C:/Program Files (x86)/Microsoft Visual Studio/2017/Community/Common7/IDE/CommonExtensions/Microsoft/CMake/CMake/share/cmake-3.12/Modules/FindBoost.cmake(959): (_Boost_COMPONENT_DEPENDENCIES)
1>   C:/Program Files (x86)/Microsoft Visual Studio/2017/Community/Common7/IDE/CommonExtensions/Microsoft/CMake/CMake/share/cmake-3.12/Modules/FindBoost.cmake(1618): (_Boost_MISSING_DEPENDENCIES)
1>   C:\test\gui/CMakeLists.txt(36): (find_package)
1> 
1> 
1> CMake Warning at C:/Program Files (x86)/Microsoft Visual Studio/2017/Community/Common7/IDE/CommonExtensions/Microsoft/CMake/CMake/share/cmake-3.12/Modules/FindBoost.cmake:847 (message):
1>   New Boost version may have incorrect or missing dependencies and imported
1>   targets
1> -- Boost version: 1.69.0
1> Call Stack (most recent call first):
1>   C:/Program Files (x86)/Microsoft Visual Studio/2017/Community/Common7/IDE/CommonExtensions/Microsoft/CMake/CMake/share/cmake-3.12/Modules/FindBoost.cmake(959): (_Boost_COMPONENT_DEPENDENCIES)
1>   C:/Program Files (x86)/Microsoft Visual Studio/2017/Community/Common7/IDE/CommonExtensions/Microsoft/CMake/CMake/share/cmake-3.12/Modules/FindBoost.cmake(1618): (_Boost_MISSING_DEPENDENCIES)
1>   C:\test\gui/CMakeLists.txt(36): (find_package)
1> 
1> 
1> -- Found the following Boost libraries:
1> --   random
1> --   system
1> -- Configuring done
1> -- Generating done
1> -- Build files have been written to: C:/Users/Donato/CMakeBuilds/0ae37d58-58e9-7033-b54d-9780727ded9c/build/x64-Release

I solved the verbosity error changin the "buildCommandArgs":"-v" to "buildCommandArgs": "-v:m"
------
But now, I have a lot of errors:
>------ Build started: Project: CMakeLists, Configuration: Release ------
c:\users\donato\.conan\data\websocketpp\0.8.1\bincrafters\stable\package\5ab84d6acfe1f23c4fae0ab88f26e3a396351ac9\include\websocketpp\frame.hpp(834): warning C4267: '=': conversion from 'size_t' to 'uint32_t', possible loss of data 
c:\users\donato\.conan\data\websocketpp\0.8.1\bincrafters\stable\package\5ab84d6acfe1f23c4fae0ab88f26e3a396351ac9\include\websocketpp\common\md5.hpp(367): warning C4267: '+=': conversion from 'size_t' to 'websocketpp::md5::md5_word_t', possible loss of data 
c:\users\donato\.conan\data\websocketpp\0.8.1\bincrafters\stable\package\5ab84d6acfe1f23c4fae0ab88f26e3a396351ac9\include\websocketpp\sha1\sha1.hpp(176): warning C4267: '=': conversion from 'size_t' to 'unsigned int', possible loss of data 
c:\test\plugin\src\logger.cpp(230): warning C4101: 'exception': unreferenced local variable 
c:\users\donato\.conan\data\websocketpp\0.8.1\bincrafters\stable\package\5ab84d6acfe1f23c4fae0ab88f26e3a396351ac9\include\websocketpp\frame.hpp(834): warning C4267: '=': conversion from 'size_t' to 'uint32_t', possible loss of data 
c:\users\donato\.conan\data\websocketpp\0.8.1\bincrafters\stable\package\5ab84d6acfe1f23c4fae0ab88f26e3a396351ac9\include\websocketpp\common\md5.hpp(367): warning C4267: '+=': conversion from 'size_t' to 'websocketpp::md5::md5_word_t', possible loss of data 
c:\users\donato\.conan\data\websocketpp\0.8.1\bincrafters\stable\package\5ab84d6acfe1f23c4fae0ab88f26e3a396351ac9\include\websocketpp\sha1\sha1.hpp(176): warning C4267: '=': conversion from 'size_t' to 'unsigned int', possible loss of data 
C:\Users\Donato\CMakeBuilds\0ae37d58-58e9-7033-b54d-9780727ded9c\build\x64-Release\gui\libssl.lib(ssl_lib.obj) : error LNK2005: SSL_CTX_check_private_key already defined in ssleay32.lib(SSLEAY32.dll) 
C:\Users\Donato\CMakeBuilds\0ae37d58-58e9-7033-b54d-9780727ded9c\build\x64-Release\gui\libssl.lib(ssl_lib.obj) : error LNK2005: SSL_CTX_ctrl already defined in ssleay32.lib(SSLEAY32.dll) 
C:\Users\Donato\CMakeBuilds\0ae37d58-58e9-7033-b54d-9780727ded9c\build\x64-Release\gui\libssl.lib(ssl_lib.obj) : error LNK2005: SSL_CTX_free already defined in ssleay32.lib(SSLEAY32.dll) 
C:\Users\Donato\CMakeBuilds\0ae37d58-58e9-7033-b54d-9780727ded9c\build\x64-Release\gui\libssl.lib(ssl_lib.obj) : error LNK2005: SSL_CTX_get_cert_store already defined in ssleay32.lib(SSLEAY32.dll) 
C:\Users\Donato\CMakeBuilds\0ae37d58-58e9-7033-b54d-9780727ded9c\build\x64-Release\gui\libssl.lib(ssl_lib.obj) : error LNK2005: SSL_CTX_load_verify_locations already defined in ssleay32.lib(SSLEAY32.dll) 
C:\Users\Donato\CMakeBuilds\0ae37d58-58e9-7033-b54d-9780727ded9c\build\x64-Release\gui\libssl.lib(ssl_lib.obj) : error LNK2005: SSL_CTX_new already defined in ssleay32.lib(SSLEAY32.dll) 
C:\Users\Donato\CMakeBuilds\0ae37d58-58e9-7033-b54d-9780727ded9c\build\x64-Release\gui\libssl.lib(ssl_lib.obj) : error LNK2005: SSL_CTX_set_alpn_protos already defined in ssleay32.lib(SSLEAY32.dll) 
C:\Users\Donato\CMakeBuilds\0ae37d58-58e9-7033-b54d-9780727ded9c\build\x64-Release\gui\libssl.lib(ssl_lib.obj) : error LNK2005: SSL_CTX_set_cipher_list already defined in ssleay32.lib(SSLEAY32.dll) 
C:\Users\Donato\CMakeBuilds\0ae37d58-58e9-7033-b54d-9780727ded9c\build\x64-Release\gui\libssl.lib(ssl_lib.obj) : error LNK2005: SSL_CTX_set_default_passwd_cb already defined in ssleay32.lib(SSLEAY32.dll) 
C:\Users\Donato\CMakeBuilds\0ae37d58-58e9-7033-b54d-9780727ded9c\build\x64-Release\gui\libssl.lib(ssl_lib.obj) : error LNK2005: SSL_CTX_set_default_passwd_cb_userdata already defined in ssleay32.lib(SSLEAY32.dll) 
C:\Users\Donato\CMakeBuilds\0ae37d58-58e9-7033-b54d-9780727ded9c\build\x64-Release\gui\libssl.lib(ssl_lib.obj) : error LNK2005: SSL_CTX_set_msg_callback already defined in ssleay32.lib(SSLEAY32.dll) 
C:\Users\Donato\CMakeBuilds\0ae37d58-58e9-7033-b54d-9780727ded9c\build\x64-Release\gui\libssl.lib(ssl_lib.obj) : error LNK2005: SSL_CTX_set_next_proto_select_cb already defined in ssleay32.lib(SSLEAY32.dll) 
C:\Users\Donato\CMakeBuilds\0ae37d58-58e9-7033-b54d-9780727ded9c\build\x64-Release\gui\libssl.lib(ssl_lib.obj) : error LNK2005: SSL_CTX_set_verify already defined in ssleay32.lib(SSLEAY32.dll) 
C:\Users\Donato\CMakeBuilds\0ae37d58-58e9-7033-b54d-9780727ded9c\build\x64-Release\gui\libssl.lib(ssl_lib.obj) : error LNK2005: SSL_connect already defined in ssleay32.lib(SSLEAY32.dll) 
C:\Users\Donato\CMakeBuilds\0ae37d58-58e9-7033-b54d-9780727ded9c\build\x64-Release\gui\libssl.lib(ssl_lib.obj) : error LNK2005: SSL_ctrl already defined in ssleay32.lib(SSLEAY32.dll) 
C:\Users\Donato\CMakeBuilds\0ae37d58-58e9-7033-b54d-9780727ded9c\build\x64-Release\gui\libssl.lib(ssl_lib.obj) : error LNK2005: SSL_free already defined in ssleay32.lib(SSLEAY32.dll) 
C:\Users\Donato\CMakeBuilds\0ae37d58-58e9-7033-b54d-9780727ded9c\build\x64-Release\gui\libssl.lib(ssl_lib.obj) : error LNK2005: SSL_get0_alpn_selected already defined in ssleay32.lib(SSLEAY32.dll) 
C:\Users\Donato\CMakeBuilds\0ae37d58-58e9-7033-b54d-9780727ded9c\build\x64-Release\gui\libssl.lib(ssl_lib.obj) : error LNK2005: SSL_get_certificate already defined in ssleay32.lib(SSLEAY32.dll) 
C:\Users\Donato\CMakeBuilds\0ae37d58-58e9-7033-b54d-9780727ded9c\build\x64-Release\gui\libssl.lib(ssl_lib.obj) : error LNK2005: SSL_get_current_cipher already defined in ssleay32.lib(SSLEAY32.dll) 
C:\Users\Donato\CMakeBuilds\0ae37d58-58e9-7033-b54d-9780727ded9c\build\x64-Release\gui\libssl.lib(ssl_lib.obj) : error LNK2005: SSL_get_error already defined in ssleay32.lib(SSLEAY32.dll) 
C:\Users\Donato\CMakeBuilds\0ae37d58-58e9-7033-b54d-9780727ded9c\build\x64-Release\gui\libssl.lib(ssl_lib.obj) : error LNK2005: SSL_get_ex_data already defined in ssleay32.lib(SSLEAY32.dll) 
C:\Users\Donato\CMakeBuilds\0ae37d58-58e9-7033-b54d-9780727ded9c\build\x64-Release\gui\libssl.lib(ssl_lib.obj) : error LNK2005: SSL_get_peer_cert_chain already defined in ssleay32.lib(SSLEAY32.dll) 
C:\Users\Donato\CMakeBuilds\0ae37d58-58e9-7033-b54d-9780727ded9c\build\x64-Release\gui\libssl.lib(ssl_lib.obj) : error LNK2005: SSL_get_peer_certificate already defined in ssleay32.lib(SSLEAY32.dll) 
C:\Users\Donato\CMakeBuilds\0ae37d58-58e9-7033-b54d-9780727ded9c\build\x64-Release\gui\libssl.lib(ssl_lib.obj) : error LNK2005: SSL_get_privatekey already defined in ssleay32.lib(SSLEAY32.dll) 
C:\Users\Donato\CMakeBuilds\0ae37d58-58e9-7033-b54d-9780727ded9c\build\x64-Release\gui\libssl.lib(ssl_lib.obj) : error LNK2005: SSL_get_verify_result already defined in ssleay32.lib(SSLEAY32.dll) 
C:\Users\Donato\CMakeBuilds\0ae37d58-58e9-7033-b54d-9780727ded9c\build\x64-Release\gui\libssl.lib(ssl_lib.obj) : error LNK2005: SSL_new already defined in ssleay32.lib(SSLEAY32.dll) 
C:\Users\Donato\CMakeBuilds\0ae37d58-58e9-7033-b54d-9780727ded9c\build\x64-Release\gui\libssl.lib(ssl_lib.obj) : error LNK2005: SSL_pending already defined in ssleay32.lib(SSLEAY32.dll) 
C:\Users\Donato\CMakeBuilds\0ae37d58-58e9-7033-b54d-9780727ded9c\build\x64-Release\gui\libssl.lib(ssl_lib.obj) : error LNK2005: SSL_read already defined in ssleay32.lib(SSLEAY32.dll) 
C:\Users\Donato\CMakeBuilds\0ae37d58-58e9-7033-b54d-9780727ded9c\build\x64-Release\gui\libssl.lib(ssl_lib.obj) : error LNK2005: SSL_set_bio already defined in ssleay32.lib(SSLEAY32.dll) 
C:\Users\Donato\CMakeBuilds\0ae37d58-58e9-7033-b54d-9780727ded9c\build\x64-Release\gui\libssl.lib(ssl_lib.obj) : error LNK2005: SSL_set_connect_state already defined in ssleay32.lib(SSLEAY32.dll) 
C:\Users\Donato\CMakeBuilds\0ae37d58-58e9-7033-b54d-9780727ded9c\build\x64-Release\gui\libssl.lib(ssl_lib.obj) : error LNK2005: SSL_set_ex_data already defined in ssleay32.lib(SSLEAY32.dll) 
C:\Users\Donato\CMakeBuilds\0ae37d58-58e9-7033-b54d-9780727ded9c\build\x64-Release\gui\libssl.lib(ssl_lib.obj) : error LNK2005: SSL_set_fd already defined in ssleay32.lib(SSLEAY32.dll) 
C:\Users\Donato\CMakeBuilds\0ae37d58-58e9-7033-b54d-9780727ded9c\build\x64-Release\gui\libssl.lib(ssl_lib.obj) : error LNK2005: SSL_shutdown already defined in ssleay32.lib(SSLEAY32.dll) 
C:\Users\Donato\CMakeBuilds\0ae37d58-58e9-7033-b54d-9780727ded9c\build\x64-Release\gui\libssl.lib(ssl_lib.obj) : error LNK2005: SSL_version already defined in ssleay32.lib(SSLEAY32.dll) 
C:\Users\Donato\CMakeBuilds\0ae37d58-58e9-7033-b54d-9780727ded9c\build\x64-Release\gui\libssl.lib(ssl_lib.obj) : error LNK2005: SSL_write already defined in ssleay32.lib(SSLEAY32.dll) 
C:\Users\Donato\CMakeBuilds\0ae37d58-58e9-7033-b54d-9780727ded9c\build\x64-Release\gui\libssl.lib(ssl_ciph.obj) : error LNK2005: SSL_CIPHER_get_name already defined in ssleay32.lib(SSLEAY32.dll) 
C:\Users\Donato\CMakeBuilds\0ae37d58-58e9-7033-b54d-9780727ded9c\build\x64-Release\gui\libssl.lib(ssl_sess.obj) : error LNK2005: SSL_CTX_sess_set_new_cb already defined in ssleay32.lib(SSLEAY32.dll) 
C:\Users\Donato\CMakeBuilds\0ae37d58-58e9-7033-b54d-9780727ded9c\build\x64-Release\gui\libssl.lib(ssl_sess.obj) : error LNK2005: SSL_SESSION_free already defined in ssleay32.lib(SSLEAY32.dll) 
C:\Users\Donato\CMakeBuilds\0ae37d58-58e9-7033-b54d-9780727ded9c\build\x64-Release\gui\libssl.lib(ssl_sess.obj) : error LNK2005: SSL_set_session already defined in ssleay32.lib(SSLEAY32.dll) 
C:\Users\Donato\CMakeBuilds\0ae37d58-58e9-7033-b54d-9780727ded9c\build\x64-Release\gui\libssl.lib(ssl_cert.obj) : error LNK2005: SSL_CTX_add_client_CA already defined in ssleay32.lib(SSLEAY32.dll) 
C:\Users\Donato\CMakeBuilds\0ae37d58-58e9-7033-b54d-9780727ded9c\build\x64-Release\gui\libssl.lib(ssl_rsa.obj) : error LNK2005: SSL_CTX_use_PrivateKey already defined in ssleay32.lib(SSLEAY32.dll) 
C:\Users\Donato\CMakeBuilds\0ae37d58-58e9-7033-b54d-9780727ded9c\build\x64-Release\gui\libssl.lib(ssl_rsa.obj) : error LNK2005: SSL_CTX_use_PrivateKey_file already defined in ssleay32.lib(SSLEAY32.dll) 
C:\Users\Donato\CMakeBuilds\0ae37d58-58e9-7033-b54d-9780727ded9c\build\x64-Release\gui\libssl.lib(ssl_rsa.obj) : error LNK2005: SSL_CTX_use_certificate already defined in ssleay32.lib(SSLEAY32.dll) 
C:\Users\Donato\CMakeBuilds\0ae37d58-58e9-7033-b54d-9780727ded9c\build\x64-Release\gui\libssl.lib(ssl_rsa.obj) : error LNK2005: SSL_CTX_use_certificate_chain_file already defined in ssleay32.lib(SSLEAY32.dll) 
C:\Users\Donato\CMakeBuilds\0ae37d58-58e9-7033-b54d-9780727ded9c\build\x64-Release\gui\libssl.lib(ssl_rsa.obj) : error LNK2005: SSL_CTX_use_certificate_file already defined in ssleay32.lib(SSLEAY32.dll) 
C:\Users\Donato\CMakeBuilds\0ae37d58-58e9-7033-b54d-9780727ded9c\build\x64-Release\gui\libcrypto.lib(asn1_lib.obj) : error LNK2005: ASN1_STRING_free already defined in libeay32.lib(LIBEAY32.dll) 
C:\Users\Donato\CMakeBuilds\0ae37d58-58e9-7033-b54d-9780727ded9c\build\x64-Release\gui\libcrypto.lib(asn1_lib.obj) : error LNK2005: ASN1_STRING_length already defined in libeay32.lib(LIBEAY32.dll) 
C:\Users\Donato\CMakeBuilds\0ae37d58-58e9-7033-b54d-9780727ded9c\build\x64-Release\gui\libcrypto.lib(asn1_lib.obj) : error LNK2005: ASN1_STRING_new already defined in libeay32.lib(LIBEAY32.dll) 
C:\Users\Donato\CMakeBuilds\0ae37d58-58e9-7033-b54d-9780727ded9c\build\x64-Release\gui\libcrypto.lib(asn1_lib.obj) : error LNK2005: ASN1_STRING_type already defined in libeay32.lib(LIBEAY32.dll) 
C:\Users\Donato\CMakeBuilds\0ae37d58-58e9-7033-b54d-9780727ded9c\build\x64-Release\gui\libcrypto.lib(digest.obj) : error LNK2005: EVP_DigestFinal_ex already defined in libeay32.lib(LIBEAY32.dll) 
C:\Users\Donato\CMakeBuilds\0ae37d58-58e9-7033-b54d-9780727ded9c\build\x64-Release\gui\libcrypto.lib(digest.obj) : error LNK2005: EVP_DigestInit_ex already defined in libeay32.lib(LIBEAY32.dll) 
C:\Users\Donato\CMakeBuilds\0ae37d58-58e9-7033-b54d-9780727ded9c\build\x64-Release\gui\libcrypto.lib(digest.obj) : error LNK2005: EVP_DigestUpdate already defined in libeay32.lib(LIBEAY32.dll) 
C:\Users\Donato\CMakeBuilds\0ae37d58-58e9-7033-b54d-9780727ded9c\build\x64-Release\gui\libcrypto.lib(p_lib.obj) : error LNK2005: EVP_PKEY_copy_parameters already defined in libeay32.lib(LIBEAY32.dll) 
C:\Users\Donato\CMakeBuilds\0ae37d58-58e9-7033-b54d-9780727ded9c\build\x64-Release\gui\libcrypto.lib(p_lib.obj) : error LNK2005: EVP_PKEY_free already defined in libeay32.lib(LIBEAY32.dll) 
C:\Users\Donato\CMakeBuilds\0ae37d58-58e9-7033-b54d-9780727ded9c\build\x64-Release\gui\libcrypto.lib(p_lib.obj) : error LNK2005: EVP_PKEY_get1_RSA already defined in libeay32.lib(LIBEAY32.dll) 
C:\Users\Donato\CMakeBuilds\0ae37d58-58e9-7033-b54d-9780727ded9c\build\x64-Release\gui\libcrypto.lib(p_lib.obj) : error LNK2005: EVP_PKEY_id already defined in libeay32.lib(LIBEAY32.dll) 
C:\Users\Donato\CMakeBuilds\0ae37d58-58e9-7033-b54d-9780727ded9c\build\x64-Release\gui\libcrypto.lib(rsa_lib.obj) : error LNK2005: RSA_free already defined in libeay32.lib(LIBEAY32.dll) 
C:\Users\Donato\CMakeBuilds\0ae37d58-58e9-7033-b54d-9780727ded9c\build\x64-Release\gui\libcrypto.lib(ex_data.obj) : error LNK2005: CRYPTO_get_ex_new_index already defined in libeay32.lib(LIBEAY32.dll) 
C:\Users\Donato\CMakeBuilds\0ae37d58-58e9-7033-b54d-9780727ded9c\build\x64-Release\gui\libcrypto.lib(mem.obj) : error LNK2005: CRYPTO_malloc already defined in libeay32.lib(LIBEAY32.dll) 
C:\Users\Donato\CMakeBuilds\0ae37d58-58e9-7033-b54d-9780727ded9c\build\x64-Release\gui\libcrypto.lib(mem.obj) : error LNK2005: CRYPTO_free already defined in libeay32.lib(LIBEAY32.dll) 
C:\Users\Donato\CMakeBuilds\0ae37d58-58e9-7033-b54d-9780727ded9c\build\x64-Release\gui\libcrypto.lib(bio_lib.obj) : error LNK2005: BIO_ctrl already defined in libeay32.lib(LIBEAY32.dll) 
C:\Users\Donato\CMakeBuilds\0ae37d58-58e9-7033-b54d-9780727ded9c\build\x64-Release\gui\libcrypto.lib(bio_lib.obj) : error LNK2005: BIO_free already defined in libeay32.lib(LIBEAY32.dll) 
C:\Users\Donato\CMakeBuilds\0ae37d58-58e9-7033-b54d-9780727ded9c\build\x64-Release\gui\libcrypto.lib(bio_lib.obj) : error LNK2005: BIO_new already defined in libeay32.lib(LIBEAY32.dll) 
C:\Users\Donato\CMakeBuilds\0ae37d58-58e9-7033-b54d-9780727ded9c\build\x64-Release\gui\libcrypto.lib(bio_lib.obj) : error LNK2005: BIO_puts already defined in libeay32.lib(LIBEAY32.dll) 
C:\Users\Donato\CMakeBuilds\0ae37d58-58e9-7033-b54d-9780727ded9c\build\x64-Release\gui\libcrypto.lib(err.obj) : error LNK2005: ERR_clear_error already defined in libeay32.lib(LIBEAY32.dll) 
C:\Users\Donato\CMakeBuilds\0ae37d58-58e9-7033-b54d-9780727ded9c\build\x64-Release\gui\libcrypto.lib(err.obj) : error LNK2005: ERR_error_string_n already defined in libeay32.lib(LIBEAY32.dll) 
C:\Users\Donato\CMakeBuilds\0ae37d58-58e9-7033-b54d-9780727ded9c\build\x64-Release\gui\libcrypto.lib(err.obj) : error LNK2005: ERR_get_error already defined in libeay32.lib(LIBEAY32.dll) 
C:\Users\Donato\CMakeBuilds\0ae37d58-58e9-7033-b54d-9780727ded9c\build\x64-Release\gui\libcrypto.lib(err.obj) : error LNK2005: ERR_peek_error already defined in libeay32.lib(LIBEAY32.dll) 
C:\Users\Donato\CMakeBuilds\0ae37d58-58e9-7033-b54d-9780727ded9c\build\x64-Release\gui\libcrypto.lib(x509_lu.obj) : error LNK2005: X509_STORE_add_lookup already defined in libeay32.lib(LIBEAY32.dll) 
C:\Users\Donato\CMakeBuilds\0ae37d58-58e9-7033-b54d-9780727ded9c\build\x64-Release\gui\libcrypto.lib(x509_lu.obj) : error LNK2005: X509_STORE_set_flags already defined in libeay32.lib(LIBEAY32.dll) 
C:\Users\Donato\CMakeBuilds\0ae37d58-58e9-7033-b54d-9780727ded9c\build\x64-Release\gui\libcrypto.lib(x_pubkey.obj) : error LNK2005: i2d_X509_PUBKEY already defined in libeay32.lib(LIBEAY32.dll) 
C:\Users\Donato\CMakeBuilds\0ae37d58-58e9-7033-b54d-9780727ded9c\build\x64-Release\gui\libcrypto.lib(x_x509.obj) : error LNK2005: X509_free already defined in libeay32.lib(LIBEAY32.dll) 
C:\Users\Donato\CMakeBuilds\0ae37d58-58e9-7033-b54d-9780727ded9c\build\x64-Release\gui\libcrypto.lib(x_x509.obj) : error LNK2005: X509_get0_signature already defined in libeay32.lib(LIBEAY32.dll) 
C:\Users\Donato\CMakeBuilds\0ae37d58-58e9-7033-b54d-9780727ded9c\build\x64-Release\gui\libcrypto.lib(x509_cmp.obj) : error LNK2005: X509_get_issuer_name already defined in libeay32.lib(LIBEAY32.dll) 
C:\Users\Donato\CMakeBuilds\0ae37d58-58e9-7033-b54d-9780727ded9c\build\x64-Release\gui\libcrypto.lib(x509_cmp.obj) : error LNK2005: X509_get_pubkey already defined in libeay32.lib(LIBEAY32.dll) 
C:\Users\Donato\CMakeBuilds\0ae37d58-58e9-7033-b54d-9780727ded9c\build\x64-Release\gui\libcrypto.lib(x509_cmp.obj) : error LNK2005: X509_get_serialNumber already defined in libeay32.lib(LIBEAY32.dll) 
C:\Users\Donato\CMakeBuilds\0ae37d58-58e9-7033-b54d-9780727ded9c\build\x64-Release\gui\libcrypto.lib(x509_cmp.obj) : error LNK2005: X509_get_subject_name already defined in libeay32.lib(LIBEAY32.dll) 
C:\Users\Donato\CMakeBuilds\0ae37d58-58e9-7033-b54d-9780727ded9c\build\x64-Release\gui\libcrypto.lib(v3_purp.obj) : error LNK2005: X509_check_issued already defined in libeay32.lib(LIBEAY32.dll) 
C:\Users\Donato\CMakeBuilds\0ae37d58-58e9-7033-b54d-9780727ded9c\build\x64-Release\gui\libcrypto.lib(rand_lib.obj) : error LNK2005: RAND_add already defined in libeay32.lib(LIBEAY32.dll) 
C:\Users\Donato\CMakeBuilds\0ae37d58-58e9-7033-b54d-9780727ded9c\build\x64-Release\gui\libcrypto.lib(rand_lib.obj) : error LNK2005: RAND_bytes already defined in libeay32.lib(LIBEAY32.dll) 
C:\Users\Donato\CMakeBuilds\0ae37d58-58e9-7033-b54d-9780727ded9c\build\x64-Release\gui\libcrypto.lib(rand_lib.obj) : error LNK2005: RAND_status already defined in libeay32.lib(LIBEAY32.dll) 
C:\Users\Donato\CMakeBuilds\0ae37d58-58e9-7033-b54d-9780727ded9c\build\x64-Release\gui\libcrypto.lib(ocsp_asn.obj) : error LNK2005: OCSP_BASICRESP_free already defined in libeay32.lib(LIBEAY32.dll) 
C:\Users\Donato\CMakeBuilds\0ae37d58-58e9-7033-b54d-9780727ded9c\build\x64-Release\gui\libcrypto.lib(ocsp_asn.obj) : error LNK2005: OCSP_RESPONSE_free already defined in libeay32.lib(LIBEAY32.dll) 
C:\Users\Donato\CMakeBuilds\0ae37d58-58e9-7033-b54d-9780727ded9c\build\x64-Release\gui\libcrypto.lib(ocsp_asn.obj) : error LNK2005: d2i_OCSP_RESPONSE already defined in libeay32.lib(LIBEAY32.dll) 
C:\Users\Donato\CMakeBuilds\0ae37d58-58e9-7033-b54d-9780727ded9c\build\x64-Release\gui\libcrypto.lib(b_print.obj) : error LNK2005: BIO_printf already defined in libeay32.lib(LIBEAY32.dll) 
C:\Users\Donato\CMakeBuilds\0ae37d58-58e9-7033-b54d-9780727ded9c\build\x64-Release\gui\libcrypto.lib(conf_mod.obj) : error LNK2005: CONF_modules_load_file already defined in libeay32.lib(LIBEAY32.dll) 
C:\Users\Donato\CMakeBuilds\0ae37d58-58e9-7033-b54d-9780727ded9c\build\x64-Release\gui\libcrypto.lib(eng_pkey.obj) : error LNK2005: ENGINE_load_private_key already defined in libeay32.lib(LIBEAY32.dll) 
C:\Users\Donato\CMakeBuilds\0ae37d58-58e9-7033-b54d-9780727ded9c\build\x64-Release\gui\libcrypto.lib(m_sha1.obj) : error LNK2005: EVP_sha256 already defined in libeay32.lib(LIBEAY32.dll) 
C:\Users\Donato\CMakeBuilds\0ae37d58-58e9-7033-b54d-9780727ded9c\build\x64-Release\gui\libcrypto.lib(rsa_crpt.obj) : error LNK2005: RSA_flags already defined in libeay32.lib(LIBEAY32.dll) 
C:\Users\Donato\CMakeBuilds\0ae37d58-58e9-7033-b54d-9780727ded9c\build\x64-Release\gui\libcrypto.lib(md5_dgst.obj) : error LNK2005: MD5_Final already defined in libeay32.lib(LIBEAY32.dll) 
C:\Users\Donato\CMakeBuilds\0ae37d58-58e9-7033-b54d-9780727ded9c\build\x64-Release\gui\libcrypto.lib(md5_dgst.obj) : error LNK2005: MD5_Init already defined in libeay32.lib(LIBEAY32.dll) 
C:\Users\Donato\CMakeBuilds\0ae37d58-58e9-7033-b54d-9780727ded9c\build\x64-Release\gui\libcrypto.lib(md5_dgst.obj) : error LNK2005: MD5_Update already defined in libeay32.lib(LIBEAY32.dll) 
C:\Users\Donato\CMakeBuilds\0ae37d58-58e9-7033-b54d-9780727ded9c\build\x64-Release\gui\libcrypto.lib(a_time.obj) : error LNK2005: ASN1_TIME_print already defined in libeay32.lib(LIBEAY32.dll) 
C:\Users\Donato\CMakeBuilds\0ae37d58-58e9-7033-b54d-9780727ded9c\build\x64-Release\gui\libcrypto.lib(bn_print.obj) : error LNK2005: BN_print already defined in libeay32.lib(LIBEAY32.dll) 
C:\Users\Donato\CMakeBuilds\0ae37d58-58e9-7033-b54d-9780727ded9c\build\x64-Release\gui\libcrypto.lib(a_object.obj) : error LNK2005: i2a_ASN1_OBJECT already defined in libeay32.lib(LIBEAY32.dll) 
C:\Users\Donato\CMakeBuilds\0ae37d58-58e9-7033-b54d-9780727ded9c\build\x64-Release\gui\libcrypto.lib(a_object.obj) : error LNK2005: i2t_ASN1_OBJECT already defined in libeay32.lib(LIBEAY32.dll) 
C:\Users\Donato\CMakeBuilds\0ae37d58-58e9-7033-b54d-9780727ded9c\build\x64-Release\gui\libcrypto.lib(by_file.obj) : error LNK2005: X509_LOOKUP_file already defined in libeay32.lib(LIBEAY32.dll) 
C:\Users\Donato\CMakeBuilds\0ae37d58-58e9-7033-b54d-9780727ded9c\build\x64-Release\gui\libcrypto.lib(by_file.obj) : error LNK2005: X509_load_crl_file already defined in libeay32.lib(LIBEAY32.dll) 
C:\Users\Donato\CMakeBuilds\0ae37d58-58e9-7033-b54d-9780727ded9c\build\x64-Release\gui\libcrypto.lib(x509_ext.obj) : error LNK2005: X509_get_ext_d2i already defined in libeay32.lib(LIBEAY32.dll) 
C:\Users\Donato\CMakeBuilds\0ae37d58-58e9-7033-b54d-9780727ded9c\build\x64-Release\gui\libcrypto.lib(x509_v3.obj) : error LNK2005: X509_EXTENSION_get_data already defined in libeay32.lib(LIBEAY32.dll) 
C:\Users\Donato\CMakeBuilds\0ae37d58-58e9-7033-b54d-9780727ded9c\build\x64-Release\gui\libcrypto.lib(x509_v3.obj) : error LNK2005: X509_EXTENSION_get_object already defined in libeay32.lib(LIBEAY32.dll) 
C:\Users\Donato\CMakeBuilds\0ae37d58-58e9-7033-b54d-9780727ded9c\build\x64-Release\gui\libcrypto.lib(v3_genn.obj) : error LNK2005: GENERAL_NAMES_free already defined in libeay32.lib(LIBEAY32.dll) 
C:\Users\Donato\CMakeBuilds\0ae37d58-58e9-7033-b54d-9780727ded9c\build\x64-Release\gui\libcrypto.lib(bss_mem.obj) : error LNK2005: BIO_s_mem already defined in libeay32.lib(LIBEAY32.dll) 
C:\Users\Donato\CMakeBuilds\0ae37d58-58e9-7033-b54d-9780727ded9c\build\x64-Release\gui\libcrypto.lib(eng_lib.obj) : error LNK2005: ENGINE_free already defined in libeay32.lib(LIBEAY32.dll) 
C:\Users\Donato\CMakeBuilds\0ae37d58-58e9-7033-b54d-9780727ded9c\build\x64-Release\gui\libcrypto.lib(eng_lib.obj) : error LNK2005: ENGINE_get_id already defined in libeay32.lib(LIBEAY32.dll) 
C:\Users\Donato\CMakeBuilds\0ae37d58-58e9-7033-b54d-9780727ded9c\build\x64-Release\gui\libcrypto.lib(x509name.obj) : error LNK2005: X509_NAME_ENTRY_get_data already defined in libeay32.lib(LIBEAY32.dll) 
C:\Users\Donato\CMakeBuilds\0ae37d58-58e9-7033-b54d-9780727ded9c\build\x64-Release\gui\libcrypto.lib(x509name.obj) : error LNK2005: X509_NAME_get_entry already defined in libeay32.lib(LIBEAY32.dll) 
C:\Users\Donato\CMakeBuilds\0ae37d58-58e9-7033-b54d-9780727ded9c\build\x64-Release\gui\libcrypto.lib(x509name.obj) : error LNK2005: X509_NAME_get_index_by_NID already defined in libeay32.lib(LIBEAY32.dll) 
C:\Users\Donato\CMakeBuilds\0ae37d58-58e9-7033-b54d-9780727ded9c\build\x64-Release\gui\libcrypto.lib(eng_fat.obj) : error LNK2005: ENGINE_set_default already defined in libeay32.lib(LIBEAY32.dll) 
C:\Users\Donato\CMakeBuilds\0ae37d58-58e9-7033-b54d-9780727ded9c\build\x64-Release\gui\libcrypto.lib(set_key.obj) : error LNK2005: DES_set_key already defined in libeay32.lib(LIBEAY32.dll) 
C:\Users\Donato\CMakeBuilds\0ae37d58-58e9-7033-b54d-9780727ded9c\build\x64-Release\gui\libcrypto.lib(set_key.obj) : error LNK2005: DES_set_odd_parity already defined in libeay32.lib(LIBEAY32.dll) 
C:\Users\Donato\CMakeBuilds\0ae37d58-58e9-7033-b54d-9780727ded9c\build\x64-Release\gui\libcrypto.lib(bss_file.obj) : error LNK2005: BIO_s_file already defined in libeay32.lib(LIBEAY32.dll) 
C:\Users\Donato\CMakeBuilds\0ae37d58-58e9-7033-b54d-9780727ded9c\build\x64-Release\gui\libcrypto.lib(p12_crpt.obj) : error LNK2005: PKCS12_PBE_add already defined in libeay32.lib(LIBEAY32.dll) 
C:\Users\Donato\CMakeBuilds\0ae37d58-58e9-7033-b54d-9780727ded9c\build\x64-Release\gui\libcrypto.lib(t_x509.obj) : error LNK2005: X509_signature_print already defined in libeay32.lib(LIBEAY32.dll) 
C:\Users\Donato\CMakeBuilds\0ae37d58-58e9-7033-b54d-9780727ded9c\build\x64-Release\gui\libcrypto.lib(ui_lib.obj) : error LNK2005: UI_create_method already defined in libeay32.lib(LIBEAY32.dll) 
C:\Users\Donato\CMakeBuilds\0ae37d58-58e9-7033-b54d-9780727ded9c\build\x64-Release\gui\libcrypto.lib(ui_lib.obj) : error LNK2005: UI_destroy_method already defined in libeay32.lib(LIBEAY32.dll) 
C:\Users\Donato\CMakeBuilds\0ae37d58-58e9-7033-b54d-9780727ded9c\build\x64-Release\gui\libcrypto.lib(ui_lib.obj) : error LNK2005: UI_get0_user_data already defined in libeay32.lib(LIBEAY32.dll) 
C:\Users\Donato\CMakeBuilds\0ae37d58-58e9-7033-b54d-9780727ded9c\build\x64-Release\gui\libcrypto.lib(ui_lib.obj) : error LNK2005: UI_get_input_flags already defined in libeay32.lib(LIBEAY32.dll) 
C:\Users\Donato\CMakeBuilds\0ae37d58-58e9-7033-b54d-9780727ded9c\build\x64-Release\gui\libcrypto.lib(ui_lib.obj) : error LNK2005: UI_get_string_type already defined in libeay32.lib(LIBEAY32.dll) 
C:\Users\Donato\CMakeBuilds\0ae37d58-58e9-7033-b54d-9780727ded9c\build\x64-Release\gui\libcrypto.lib(ui_lib.obj) : error LNK2005: UI_method_get_closer already defined in libeay32.lib(LIBEAY32.dll) 
C:\Users\Donato\CMakeBuilds\0ae37d58-58e9-7033-b54d-9780727ded9c\build\x64-Release\gui\libcrypto.lib(ui_lib.obj) : error LNK2005: UI_method_get_opener already defined in libeay32.lib(LIBEAY32.dll) 
C:\Users\Donato\CMakeBuilds\0ae37d58-58e9-7033-b54d-9780727ded9c\build\x64-Release\gui\libcrypto.lib(ui_lib.obj) : error LNK2005: UI_method_get_reader already defined in libeay32.lib(LIBEAY32.dll) 
C:\Users\Donato\CMakeBuilds\0ae37d58-58e9-7033-b54d-9780727ded9c\build\x64-Release\gui\libcrypto.lib(ui_lib.obj) : error LNK2005: UI_method_get_writer already defined in libeay32.lib(LIBEAY32.dll) 
C:\Users\Donato\CMakeBuilds\0ae37d58-58e9-7033-b54d-9780727ded9c\build\x64-Release\gui\libcrypto.lib(ui_lib.obj) : error LNK2005: UI_method_set_closer already defined in libeay32.lib(LIBEAY32.dll) 
C:\Users\Donato\CMakeBuilds\0ae37d58-58e9-7033-b54d-9780727ded9c\build\x64-Release\gui\libcrypto.lib(ui_lib.obj) : error LNK2005: UI_method_set_opener already defined in libeay32.lib(LIBEAY32.dll) 
C:\Users\Donato\CMakeBuilds\0ae37d58-58e9-7033-b54d-9780727ded9c\build\x64-Release\gui\libcrypto.lib(ui_lib.obj) : error LNK2005: UI_method_set_reader already defined in libeay32.lib(LIBEAY32.dll) 
C:\Users\Donato\CMakeBuilds\0ae37d58-58e9-7033-b54d-9780727ded9c\build\x64-Release\gui\libcrypto.lib(ui_lib.obj) : error LNK2005: UI_method_set_writer already defined in libeay32.lib(LIBEAY32.dll) 
C:\Users\Donato\CMakeBuilds\0ae37d58-58e9-7033-b54d-9780727ded9c\build\x64-Release\gui\libcrypto.lib(ui_lib.obj) : error LNK2005: UI_set_result already defined in libeay32.lib(LIBEAY32.dll) 
C:\Users\Donato\CMakeBuilds\0ae37d58-58e9-7033-b54d-9780727ded9c\build\x64-Release\gui\libcrypto.lib(a_print.obj) : error LNK2005: ASN1_STRING_print already defined in libeay32.lib(LIBEAY32.dll) 
C:\Users\Donato\CMakeBuilds\0ae37d58-58e9-7033-b54d-9780727ded9c\build\x64-Release\gui\libcrypto.lib(eng_list.obj) : error LNK2005: ENGINE_by_id already defined in libeay32.lib(LIBEAY32.dll) 
C:\Users\Donato\CMakeBuilds\0ae37d58-58e9-7033-b54d-9780727ded9c\build\x64-Release\gui\libcrypto.lib(eng_list.obj) : error LNK2005: ENGINE_get_first already defined in libeay32.lib(LIBEAY32.dll) 
C:\Users\Donato\CMakeBuilds\0ae37d58-58e9-7033-b54d-9780727ded9c\build\x64-Release\gui\libcrypto.lib(eng_list.obj) : error LNK2005: ENGINE_get_next already defined in libeay32.lib(LIBEAY32.dll) 
C:\Users\Donato\CMakeBuilds\0ae37d58-58e9-7033-b54d-9780727ded9c\build\x64-Release\gui\libcrypto.lib(eng_init.obj) : error LNK2005: ENGINE_finish already defined in libeay32.lib(LIBEAY32.dll) 
C:\Users\Donato\CMakeBuilds\0ae37d58-58e9-7033-b54d-9780727ded9c\build\x64-Release\gui\libcrypto.lib(eng_init.obj) : error LNK2005: ENGINE_init already defined in libeay32.lib(LIBEAY32.dll) 
C:\Users\Donato\CMakeBuilds\0ae37d58-58e9-7033-b54d-9780727ded9c\build\x64-Release\gui\libcrypto.lib(v3_prn.obj) : error LNK2005: X509V3_EXT_print already defined in libeay32.lib(LIBEAY32.dll) 
C:\Users\Donato\CMakeBuilds\0ae37d58-58e9-7033-b54d-9780727ded9c\build\x64-Release\gui\libcrypto.lib(ui_openssl.obj) : error LNK2005: UI_OpenSSL already defined in libeay32.lib(LIBEAY32.dll) 
C:\Users\Donato\CMakeBuilds\0ae37d58-58e9-7033-b54d-9780727ded9c\build\x64-Release\gui\libcrypto.lib(eng_ctrl.obj) : error LNK2005: ENGINE_ctrl already defined in libeay32.lib(LIBEAY32.dll) 
C:\Users\Donato\CMakeBuilds\0ae37d58-58e9-7033-b54d-9780727ded9c\build\x64-Release\gui\libcrypto.lib(eng_ctrl.obj) : error LNK2005: ENGINE_ctrl_cmd already defined in libeay32.lib(LIBEAY32.dll) 
C:\Users\Donato\CMakeBuilds\0ae37d58-58e9-7033-b54d-9780727ded9c\build\x64-Release\gui\libcrypto.lib(p12_asn.obj) : error LNK2005: PKCS12_free already defined in libeay32.lib(LIBEAY32.dll) 
C:\Users\Donato\CMakeBuilds\0ae37d58-58e9-7033-b54d-9780727ded9c\build\x64-Release\gui\libcrypto.lib(a_strex.obj) : error LNK2005: ASN1_STRING_to_UTF8 already defined in libeay32.lib(LIBEAY32.dll) 
C:\Users\Donato\CMakeBuilds\0ae37d58-58e9-7033-b54d-9780727ded9c\build\x64-Release\gui\libcrypto.lib(a_strex.obj) : error LNK2005: X509_NAME_print_ex already defined in libeay32.lib(LIBEAY32.dll) 
C:\Users\Donato\CMakeBuilds\0ae37d58-58e9-7033-b54d-9780727ded9c\build\x64-Release\gui\libcrypto.lib(p12_utl.obj) : error LNK2005: d2i_PKCS12_bio already defined in libeay32.lib(LIBEAY32.dll) 
C:\Users\Donato\CMakeBuilds\0ae37d58-58e9-7033-b54d-9780727ded9c\build\x64-Release\bin\ets2-job-logger.exe : fatal error LNK1169: one or more multiply defined symbols found

--- a/.gitignore
+++ b/.gitignore
@@ -1,1 +1,3 @@
 build/
+.vs/
+CMakeSettings.json

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -3,6 +3,11 @@ project(ets2-job-logger)
 
 set(CMAKE_CXX_STANDARD 11)
 
+if(MSVC)
+    include(${CMAKE_BINARY_DIR}/conanbuildinfo.cmake)
+    conan_basic_setup()
+endif()
+
 add_subdirectory(scs-sdk)
 add_subdirectory(plugin)
 add_subdirectory(gui)

--- a/README.md
+++ b/README.md
@@ -68,3 +68,41 @@ code is `200`.
 - `openssl`
 - `jsoncpp`
 - `wxWidgets`
+
+### Windows
+
+Install `conan`. Open `PowerShell` and add remotes for `conan`:
+
+```
+conan remote add public-conan https://api.bintray.com/conan/bincrafters/public-conan
+conan remote add theirix https://api.bintray.com/conan/bincrafters/theirix
+```
+
+- Open project in Visual Studio via `File -> Open -> CMake...` by choosing the
+  `CMakeListst.xt`.
+- Edit CMake options via `CMake -> Change CMake settings -> CMakelists.txt`
+  - Change `generator` value to `Visual Studio 15 2017 Win64`
+- CMake will produce error, note down the path from the error:
+  > include could not find load file:
+  >
+  > C:/Users/Test/CMakeBuilds/d76aad98-e5ad-0738-bbe9-bc20c6d93f9c/build/x64-Debug/conanbuildinfo.cmake`
+- Open PowerShell in the source directory and run:
+  - `conan install . -s build_type=Debug -if='<PATH_FROM_PREVIOUS_STEP>'`
+  - `msgpack-c` version `2.3.0` has issue which needs an manual fix:
+    - Go to `C:\Users\<YOUR_USERNAME>\.conan\data\msgpack\3.2.0\bincrafters\stable\package`.
+      This directory should have two directories.
+    - In both directories go to `lib\cmake\msgpack`
+      and open `msgpack-config.cmake` file in your favorite text editor.
+    - Remove following lines at the bottom:
+      ```
+      if(NOT OFF)
+        add_library(msgpackc ALIAS msgpackc-static)
+      endif()
+      ```
+- After installation finished go back to Visual Studio click `Generate` in the
+  notification.
+- After CMake generation finished you're ready to build.
+
+When you're ready to build a release just replace `Debug` with `Release` in:
+- CMakeSettings.json (`CMake -> Change CMake settings -> CMakelists.txt`)
+- Replace all `Debug` parts of the `conan` install command.

--- a/README.md
+++ b/README.md
@@ -81,7 +81,8 @@ conan remote add theirix https://api.bintray.com/conan/bincrafters/theirix
 - Open project in Visual Studio via `File -> Open -> CMake...` by choosing the
   `CMakeListst.xt`.
 - Edit CMake options via `CMake -> Change CMake settings -> CMakelists.txt`
-  - Change `generator` value to `Visual Studio 15 2017 Win64`
+  - Change `generator` value to `Visual Studio 15 2017 Win64`.
+  - Remove `-v` from `buildCommandArgs` or specify valid verbosity level.
 - CMake will produce error, note down the path from the error:
   > include could not find load file:
   >

--- a/conanfile.txt
+++ b/conanfile.txt
@@ -1,0 +1,12 @@
+[requires]
+msgpack/3.2.0@bincrafters/stable
+websocketpp/0.8.1@bincrafters/stable
+libcurl/7.64.1@bincrafters/stable
+jsoncpp/1.8.4@theirix/stable
+wxwidgets/3.1.2@bincrafters/stable
+
+[generators]
+cmake
+
+[options]
+libcurl:with_openssl=True

--- a/conanfile.txt
+++ b/conanfile.txt
@@ -1,12 +1,10 @@
 [requires]
 msgpack/3.2.0@bincrafters/stable
+zlib/1.2.11@conan/stable
 websocketpp/0.8.1@bincrafters/stable
-libcurl/7.64.1@bincrafters/stable
+libcurl/7.66.0@bincrafters/stable
 jsoncpp/1.8.4@theirix/stable
 wxwidgets/3.1.2@bincrafters/stable
 
 [generators]
 cmake
-
-[options]
-libcurl:with_openssl=True

--- a/gui/CMakeLists.txt
+++ b/gui/CMakeLists.txt
@@ -29,11 +29,24 @@ set(SOURCES
     )
 
 find_package(wxWidgets 3.0 REQUIRED)
-find_package(CURL REQUIRED)
-find_package(PkgConfig REQUIRED)
-find_package(OpenSSL REQUIRED)
-pkg_check_modules(JSONCPP jsoncpp REQUIRED)
 include(${wxWidgets_USE_FILE})
+if (MSVC)
+    find_package(PNG REQUIRED)
+    set(Boost_USE_STATIC_LIBS ON)
+    find_package(Boost REQUIRED COMPONENTS random)
+endif()
+find_package(ZLIB REQUIRED)
+find_package(OpenSSL REQUIRED)
+find_package(CURL REQUIRED)
+find_package(OpenSSL REQUIRED)
+if (MSVC)
+    find_package(jsoncpp CONFIG REQUIRED)
+    set(JSONCPP_LIBRARIES jsoncpp_lib_static)
+    set(JSONCPP_INCLUDE_DIRS jsoncpp_lib_static)
+else()
+    find_package(PkgConfig REQUIRED)
+    pkg_check_modules(JSONCPP jsoncpp REQUIRED)
+endif()
 
 if (NOT WIN32)
     if (PLUGIN_PATH)
@@ -48,18 +61,28 @@ configure_file(Logger/PluginInstallerDefs.h.in PluginInstallerDefs.h)
 
 add_executable(ets2-job-logger WIN32 ${SOURCES})
 
-if (WIN32)
+if (MINGW)
     set(WIN32_LIBS -lws2_32)
+elseif (WIN32)
+    find_library(WS2_32_LIB ws2_32.lib REQUIRED)
+    find_library(GDI32_LIB gdi32.lib REQUIRED)
+    find_library(ADVAPI32_LIB advapi32.lib REQUIRED)
+    find_library(CRYPT32_LIB crypt32.lib REQUIRED)
+    find_library(USER32_LIB user32.lib REQUIRED)
+    set(WIN32_LIBS ${WS2_32_LIB} ${GDI32_LIB} ${ADVAPI32_LIB} ${CRYPT32_LIB} ${USER32_LIB})
 endif()
 
 target_link_libraries(ets2-job-logger
                       PRIVATE
                       ets2joblogger::plugin
                       ${wxWidgets_LIBRARIES}
+                      ${PNG_LIBRARIES}
+                      ${Boost_LIBRARIES}
                       ${WIN32_LIBS}
-                      ${JSONCPP_LIBRARIES}
+                      ${ZLIB_LIBRARIES}
+                      ${OPENSSL_LIBRARIES}
                       ${CURL_LIBRARIES}
-                      ${OPENSSL_CRYPTO_LIBRARY}
+                      ${JSONCPP_LIBRARIES}
                       )
 
 target_include_directories(ets2-job-logger
@@ -68,8 +91,12 @@ target_include_directories(ets2-job-logger
                            "${CMAKE_CURRENT_BINARY_DIR}"
                            "${WEBSOCKETPP_INCLUDE_DIR}"
                            "${wxWidgets_INCLUDE_DIRS}"
-                           "${JSONCPP_INCLUDE_DIRS}"
+                           "${PNG_INCLUDE_DIRS}"
+                           "${Boost_INCLUDE_DIRS}"
+                           "${ZLIB_INCLUDE_DIRS}"
+                           "${OPENSSL_INCLUDE_DIRS}"
                            "${CURL_INCLUDE_DIRS}"
+                           "${JSONCPP_INCLUDE_DIRS}"
                            )
 
 target_compile_options(ets2-job-logger

--- a/gui/Logger/PluginInstaller.cpp
+++ b/gui/Logger/PluginInstaller.cpp
@@ -120,8 +120,8 @@ wxString get_game_bin_directory(const PluginInstaller::Platform &platform,
 }
 
 wxString get_sha256(const wxString &file) {
-    std::ifstream f(file, std::ios::in | std::ios::binary);
-    if (!f.good()) {
+    wxFile f(file, wxFile::OpenMode::read);
+    if (!f.IsOpened()){
         return "";
     }
 
@@ -131,13 +131,13 @@ wxString get_sha256(const wxString &file) {
     SHA256_CTX ctx;
     SHA256_Init(&ctx);
 
-    while (f.good()) {
-        f.read(buffer, 4096);
-        SHA256_Update(&ctx, buffer, f.gcount());
+    while (!f.Eof()) {
+        auto read = f.Read(buffer, 4096);
+        SHA256_Update(&ctx, buffer, read);
     }
 
     SHA256_Final(hash, &ctx);
-    f.close();
+    f.Close();
 
     std::stringstream ss;
     ss << std::hex << std::setfill('0');

--- a/plugin/CMakeLists.txt
+++ b/plugin/CMakeLists.txt
@@ -1,7 +1,7 @@
 project(plugin VERSION 0.1.0)
 
 add_library(${PROJECT_NAME}
-            SHARED
+            OBJECT
             include/jobplugin/PluginDefs.h
 
             src/Logger.cpp
@@ -13,6 +13,8 @@ add_library(${PROJECT_NAME}
 configure_file(include/jobplugin/Version.h.in "include/jobplugin/Version.h")
 
 add_library(ets2joblogger::plugin ALIAS ${PROJECT_NAME})
+add_library(plugin_shared SHARED $<TARGET_OBJECTS:${PROJECT_NAME}>)
+add_library(plugin_static STATIC $<TARGET_OBJECTS:${PROJECT_NAME}>)
 
 find_package(msgpack REQUIRED)
 find_package(websocketpp REQUIRED)
@@ -34,7 +36,13 @@ if(MINGW)
         )
 endif()
 
-set_target_properties(${PROJECT_NAME}
+set_target_properties(plugin_shared
+                      PROPERTIES
+                      OUTPUT_NAME "ets2-job-logger"
+                      PREFIX ""
+                      )
+
+set_target_properties(plugin_static
                       PROPERTIES
                       OUTPUT_NAME "ets2-job-logger"
                       PREFIX ""
@@ -47,12 +55,11 @@ target_link_libraries(${PROJECT_NAME}
                       )
 
 target_include_directories(${PROJECT_NAME}
-                           INTERFACE
-                           "${CMAKE_CURRENT_LIST_DIR}/include"
                            PUBLIC
-                           "${CMAKE_CURRENT_BINARY_DIR}/include"
+                           $<INSTALL_INTERFACE:include>
+                           $<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}/include>
+                           $<BUILD_INTERFACE:${CMAKE_CURRENT_BINARY_DIR}/include>
                            PRIVATE
-                           "${CMAKE_CURRENT_SOURCE_DIR}/include"
                            "${WEBSOCKETPP_INCLUDE_DIR}"
                            )
 

--- a/plugin/include/jobplugin/PluginDefs.h
+++ b/plugin/include/jobplugin/PluginDefs.h
@@ -28,7 +28,7 @@
 #ifndef ETS2_JOB_LOGGER_PLUGINDEFS_H
 #define ETS2_JOB_LOGGER_PLUGINDEFS_H
 
-#include "jobplugin/Version.h"
+#include <jobplugin/Version.h>
 #include <string>
 #include <msgpack.hpp>
 #ifndef PLUGIN_INTERNAL

--- a/plugin/src/main.cpp
+++ b/plugin/src/main.cpp
@@ -27,11 +27,14 @@
 #include "Logger.h"
 #include <functional>
 #include <scs/scssdk_telemetry.h>
-#include <scs/common/scssdk_telemetry_common_configs.h>
 #include <scs/common/scssdk_telemetry_truck_common_channels.h>
 #include <scs/common/scssdk_telemetry_trailer_common_channels.h>
 #include <scs/amtrucks/scssdk_ats.h>
 #include <scs/eurotrucks2/scssdk_eut2.h>
+
+#ifndef _WIN32
+#define __stdcall
+#endif
 
 typedef std::function<SCSAPI_VOID(const scs_value_t *const)> callback_channel_t;
 typedef std::function<SCSAPI_VOID(const void *const event_info)> callback_event_t;
@@ -46,19 +49,19 @@ callback_event_t config_cb;
 
 namespace cb {
     namespace bind {
-        callback_channel_t channel(void (Logger::*func)(const scs_value_t *const), Logger *obj) {
+        callback_channel_t channel(void (__stdcall Logger::*func)(const scs_value_t *const), Logger *obj) {
             return std::bind(func, obj, std::placeholders::_1);
         }
 
-        callback_event_t event(void (Logger::*func)(const scs_telemetry_frame_start_t *const), Logger *obj) {
+        callback_event_t event(void (__stdcall Logger::*func)(const scs_telemetry_frame_start_t *const), Logger *obj) {
             return std::bind((void (Logger::*)(const void *const))(func), obj, std::placeholders::_1);
         }
 
-        callback_event_t event(void (Logger::*func)(const void *const), Logger *obj) {
+        callback_event_t event(void (__stdcall Logger::*func)(const void *const), Logger *obj) {
             return std::bind((void (Logger::*)(const void *const))(func), obj, std::placeholders::_1);
         }
 
-        callback_event_t event(void (Logger::*func)(const scs_telemetry_configuration_t *), Logger *obj) {
+        callback_event_t event(void (__stdcall Logger::*func)(const scs_telemetry_configuration_t *), Logger *obj) {
             return std::bind((void (Logger::*)(const void *const))(func), obj, std::placeholders::_1);
         }
     }

